src/components/QueryEditor/VisualQueryEditor/AggregateItem.tsx

```diff
@@ -0,0 +1,131 @@
+  Sum = 'sum',
+  Avg = 'avg',
+  Count = 'count',
+  Dcount = 'dcount',
+  Max = 'max',
+  Min = 'min',
+  Percentile = 'percentile',
+}
+
+interface AggregateItemProps {
+  datasource: AdxDataSource;
+  query: KustoQuery;
+  aggregate: Partial<QueryEditorReduceExpression>;
+  columns: AdxColumnSchema[] | undefined;
+  templateVariableOptions: SelectableValue<string>;
+  onChange: (item: QueryEditorReduceExpression) => void;
+  onDelete: () => void;
+}
+
+const AggregateItem: React.FC<AggregateItemProps> = (props) => {
+  const { aggregate, onChange, onDelete, columns, templateVariableOptions } = props;
+
+  let columnOptions: Array<SelectableValue<string>> = columns
+    ? columnsToDefinition(columns).map((c) => ({ label: c.label, value: c.value }))
+    : [];
+  columnOptions = columnOptions.concat(templateVariableOptions);
+
+  return (
+    <InputGroup>
+      <Select
+        aria-label="function"
+        width="auto"
+        value={aggregate.reduce?.name ? valueToDefinition(aggregate.reduce?.name) : null}
+        options={Object.values(AggregateFunctions).map((f) => ({ label: f, value: f }))}
+        allowCustomValue
+        onChange={(e) =>
+          e.value &&
+          onChange({
+            reduce: { name: e.value, type: QueryEditorPropertyType.Function },
+            property: {
+              name: aggregate.property?.name || '',
+              type: aggregate.property?.type || QueryEditorPropertyType.String,
+            },
+            parameters: aggregate.parameters,
+            type: QueryEditorExpressionType.Reduce,
+          })
+        }
+      />
+      {aggregate.reduce?.name === AggregateFunctions.Percentile && (
+        <Select
+          aria-label="percentile"
+          options={range(0, 100, 5).map((n) => ({ label: n.toString(), value: n.toString() }))}
+          value={aggregate.parameters?.length ? aggregate.parameters[0].value : undefined}
+          width="auto"
+          allowCustomValue
+          onChange={(e) => {
+            e.value &&
+              onChange({
+                property: {
+                  name: aggregate.property?.name || '',
+                  type: aggregate.property?.type || QueryEditorPropertyType.String,
+                },
+                reduce: {
+                  name: aggregate.reduce?.name || '',
+                  type: aggregate.property?.type || QueryEditorPropertyType.Function,
+                },
+                parameters: [
+                  {
+                    type: QueryEditorExpressionType.FunctionParameter,
+                    fieldType: QueryEditorPropertyType.Number,
+                    value: e.value,
+                    name: 'percentileParam',
+                  },
+                ],
+                type: QueryEditorExpressionType.Reduce,
+              });
+          }}
+        />
+      )}
+      {aggregate.reduce?.name !== AggregateFunctions.Count && (
+        <>
+          <Label style={{ margin: '9px 9px 0 9px' }}>of</Label>
```

Comment:
Do we have this pattern else-where? The `of` feels disconnected to me. Might be worth running by UX?